### PR TITLE
Add selected native views

### LINKED ISSUE #34: Add selected native views
The following views are found in the CRMP database. Some subset of them should be added with native view support (i.e., with support for creating and dropping views, as was added for the weather anomaly views and matviews, following this [SQLAlchemy Usage Recipe](https://github.com/sqlalchemy/sqlalchemy/wiki/Views)).

Two views, `crmp_network_geoserver` and `obs_with_flags` are already mapped by tables (with no native view support) in PyCDS. Their table mappings should be replaced with view definitions.

- [x] categorize these views as:
  - in use in production software (P)
  - generally useful (G)
  - used for analysis, report generation, but not in production code (A)
  - not used (N)
  - remove (~strikethrough~) or keep (normal)
- Reminder: This list does not include views that support the manual matviews.
- [x] `crmp_network_geoserver` (P: pdp) (K) **NB**: Already present as a table mapping. Replace.
- [x] ~`daily_agg`~ (A: Faron; ~P~)
- [x] ~`daily_view`~  (A: Faron; ~P~) used by `daily_agg`
- [x] ~`env_aqn_derived_daily_temp`~  (A/N?: Faron; ~P~)
- [x] ~`env_aqn_derived_monthly_temp`~ (A/N?: Faron; ~P~)
- [x] ~`farons_favs`~ (N)
- [x] ~`frbc_derived_daily_temp`~  (A/N?: Faron; ~P~)
- [x] ~`frbc_derived_monthly_temp`~  (A/N?: Faron; ~P~)
- [x] `history_join_station_network` (A: Faron; ~P~)
- [x] ~`history_join_station_network_geoserver`~ (?: Faron; ~P~)
- [x] ~`hourly_view`~ (A: Faron, by inference; ~P~) used by `daily_agg`
- [x] ~`mof_derived_daily_temp`~  (A/N?: Faron; ~P~)
- [x] ~`mof_derived_monthly_temp`~ (A/N?: Faron; ~P~)
- [x] ~`monthly_agg`~  (A: Faron; ~P~)
- [x] ~`monthly_agg_v`~ (A: Faron; ~P~)
- [x] `obs_count_per_day_history_v` (A: Faron; ~P~)
- [x] `obs_with_flags`  (A: Faron; P: used by fns `query_one_station`, `query_one_station_climo`) **NB**: Already present as a table mapping. Replace.
- [x] ~`obs_with_flags_history`~ (~P~; N?: Faron)
- [x] ~`seasonal_agg_v`~  (~P~; N?: Faron)

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #34 

This PR adds the following views as native SQL supported views via SQLAlchemy:

- [x] `crmp_network_geoserver` (replaces table mapping)
- [x] `history_join_station_network`
- [x] `obs_count_per_day_history_v`
- [x] `obs_with_flags` (replaces table mapping)

It also slightly improves the SQLAlchemy support for native views.

### CODE REVIEW

**Comment by rod-glover** [comment]
@jameshiebert and @faronium , would you mind weighing in on the issue described below?

The original table mapping class `ObsWithFlags` for the view `obs_with_flags` has 4 columns that are not defined in the actual view in the current CRMP database. The additional columns are `native_flag_id`, `flag_name`, `description`, and `flag_value`.

Furthermore, the docstring for the table mapping class `ObsWithFlags` doesn't entirely make sense. The relationship between native flags (table `meta_native_flag`) and observations is many-to-many, represented by the table `obs_raw_native_flags`. But the 4 additional columns in `ObsWithFlags` allow for at most one flag per observation to be represented. And the docstring states "with additional columns for each attached flag", which is (a) not true, and (b) not feasible or wise to implement in the general sense of arbitrarily many flags per observation in a single row.

Any real problems will lie with code that uses this view. Fortunately it appears this is not the case:

A [search](https://github.com/search?q=org%3Apacificclimate+in%3Afile+ObsWithFlags&type=Code) of our entire codebase on GitHub shows only one [usage of `ObsWithFlags`](https://github.com/pacificclimate/pycds/blob/eea56cadc512fa3e13c2eb83a889997b0ae61775/pycds/util.py#L49-L99). This is in PyCDS itself, and it looks outdated. The using function, `orm_station_table`, also is only used within PyCDS itself (and only in tests). So any issue related to `ObsWithFlags` is limited to PyCDS.

A [search](https://github.com/search?q=org%3Apacificclimate+in%3Afile+obs_with_flags&type=Code) for the table name `obs_with_flags` turns up references in the `crmp` repo and in PyCDS. The usages in crmp appear to be outdated code that formed the pattern for functions `query_one_station_climo` and `query_one_station_climo`, which in the current CRMP database (and the corresponding new code in PyCDS) have been reformulated without use of `obs_with_flags`.

Therefore, it looks as if the problem is more apparent than real because no production code uses it.

I *think* the correct procedure here is to go with the current definition in CRMP (now embodied in new PyCDS code in this PR).

Here is the original table mapping, with extra columns and misleading docstring.

```
class ObsWithFlags(DeferredBase):	
    '''This class maps to a convenience view that is used to construct a	
    table of flagged observations; i.e. one row per observation with	
    additional columns for each attached flag.	
    '''	
    __tablename__ = 'obs_with_flags'	
    __table_args__ = {'info': {'is_view': True}}	
    vars_id = Column(Integer, ForeignKey('meta_vars.vars_id'))	
    network_id = Column(Integer, ForeignKey('meta_network.network_id'))	
    unit = Column(String)	
    standard_name = Column(String)	
    cell_method = Column(String)	
    net_var_name = Column(String)	
    obs_raw_id = Column(Integer, ForeignKey(	
        'obs_raw.obs_raw_id'), primary_key=True)	
    station_id = Column(Integer, ForeignKey('meta_station.station_id'))	
    obs_time = Column(DateTime)	
    mod_time = Column(DateTime)	
    datum = Column(Float)	
    native_flag_id = Column(Integer, ForeignKey(	
        'meta_native_flag.native_flag_id'))	with	
    additional columns for each attached flag
    flag_name = Column(String)	
    description = Column(String)	
    flag_value = Column(String)
```

**Comment by jameshiebert** [comment]
First I'll comment that +366 −2,158 LOC is always a good sign! :)

**Comment by jameshiebert** [comment]
With respect to `obs_with_flags` it's been a *long* time since I've looked at it in close detail, and I don't have a detailed memory of what was the intent and whether the implementation ultimately met that intent.

I will say that having a *convenient* way to access the attributed flags for a given observation is a functionality that we are desperately without which has prevented at least some QA/QC work in the past.

Obviously we do not want the docstring to misrepresent the code, and if the current code doesn't actually use `ObsWithFlags`, then you're not going to break anything by bringing the definition into compliance. But do keep in mind that there is a real need for `ObsWithFlags` even if it's not presently utilized.

**Comment by rod-glover** [comment]
@jameshiebert , re. flags, see https://github.com/pacificclimate/pycds/issues/38

**Comment by jameshiebert** [comment]
Yeah, I realize that it is not feasible in the general case, but practically, we *do* do this elsewhere in the database. For example, `do_query_one_station()` returns a variable number of columns based on the number of variables a station measures.

While I don't want to set us up for making a poor assumption and then having chaos/breakage later on, there *is* value to a view like this. Even if that's not how it's written now, I would wager a guess that it was at one point.

**Comment by rod-glover** [comment]
I don't think this can be done with a view. `query_one_station` executes a query that is built per station, and that query returns a variable number of columns based on the station. This is really just quibbling, unless you are literally talking about a view that encompasses more than one station. I think the only answer to this is more or less the one embodied in the function/stored procedure `query_one_station`, which executes a pile of plpython.

TL;DR (at the end, hah): I think there's nothing being lost here, and it will be interesting to formulate the view/function/whatever that we need for the purpose you are identifying.

**Comment by rod-glover** [comment]
@jameshiebert , are you still reviewing or are we done?

**Comment by jameshiebert** [comment]
Looks good @rod-glover , sorry that this fell through the cracks. Merge away :)
